sequencer/src/state.rs

```diff
@@ -512,17 +543,71 @@ impl ValidatedState {
         } else {
             vec![]
         };
+        let accounts = std::iter::once(proposed_header.fee_info.account);
+
+        // Find missing state entries
+        let missing_accounts = self.forgotten_accounts(
+            accounts.chain(l1_deposits.iter().map(|fee_info| fee_info.account)),
+        );
+
+        let view = parent_leaf.get_view_number();
+
+        // Ensure merkle tree has frontier
+        if self.need_to_fetch_blocks_mt_frontier() {
+            tracing::warn!("fetching block frontier from peers");
+            instance
+                .peers
+                .as_ref()
+                .remember_blocks_merkle_tree(view, &mut validated_state.block_merkle_tree)
+                .await
+                .unwrap();
+
+            // validate proposal is descendent of parent by appending to parent
```

Comment:
This comment is now outdated, we're not validating anything here, we're just appending.

Reply:
Also we need to do this outside of this `if` statement, whether or not we had to fetch the frontier from peers is irrelevant, we need to insert the new block regardless